### PR TITLE
Add a development mode to catch files being deleted or moved

### PR DESCRIPTION
**Pull request by casperisfine**
As discussed. In development, it can happen that a requirable file is deleted or moved from a load path entry to another.

This fix isn't perfect, there is still edge cases that can trick it (as outlines in the pending test), but it should still be better than what we have now. 

@kirs for review please.

### CODE REVIEW

**Review by kirs — APPROVED** [review]
👏 👏 👏

File: lib/bootscale/cache_builder.rb [hunk 1]
```diff
@@ -4,13 +4,21 @@ def initialize
...
 
+    def clear!
+      @entries = {}
+    end
+
     # generate the requireables cache from all current load-path entries
     # each load-path is cached individually, so new ones can be added or removed
     # but added/removed files will not be discovered
-    def generate(load_path)
+    def generate(load_path, rescan = false)
```
Comment (grosser) [inline lib/bootscale/cache_builder.rb:14]: prefer keyword args for readability

File: spec/cache_spec.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    Bootscale.cache_builder.clear!
+    files.each do |relative_path|
+      absolute_path = File.join(tmpdir, relative_path)
+      FileUtils.mkdir_p(File.dirname(absolute_path))
+      File.write(absolute_path, '')
+    end
+  end
+
+  after do
+    FileUtils.rm_rf(tmpdir)
```
Comment (grosser) [inline spec/cache_spec.rb:27]: would be easier to follow as an `around` that creates and cleanes up
